src/utils/request.js

```diff
@@ -1,7 +1,7 @@
 import {defer, isXml, parse} from "./core";
 import Path from "./path";
 
-function request(url, type, withCredentials, headers) {
+function request(url, type, withCredentials, headers, returnResponseObj) {
```

Comment:
I'd like the address this xml issue in a separate pull request.
I'm thinking we  can use the type that is passed to determine if the responseXML should be used.